File: WEB/app/component/favorites/favorites.dart
import 'package:angular2/core.dart';
import 'package:angular2/router.dart';

@Component(
    selector: 'in-tuition-favorites',
    templateUrl: 'favorites.html',
    styleUrls: const ['favorites.css'],
    directives: const [],
    providers: const []
)

class FavoritesComponent{
  final Router _router;

  FavoritesComponent(this._router);
}

File: WEB/app/component/favorites/favorites.ts
/**
 * Created by praghav on 5/14/2016.
 */

import {Component,OnInit, OnDestroy} from '@angular/core';

@Component({
    selector: 'in-tuition-favorites',
    styleUrls: ['app/components/favorites.css'],
    templateUrl:'app/components/favorites.html',
    directives: [],
    providers: []
})

export class FavoritesComponent implements OnInit, OnDestroy{

    constructor(){

    }

    ngOnInit() {
        console.log('ngOnInit');
    }
    ngOnDestroy() {
        console.log('ngOnDestroy');
    }
}